test/src/unittests/tonal/test_audio2midi.py

```diff
@@ -0,0 +1,312 @@
+#!/usr/bin/env python
+
+# Copyright (C) 2006-2024  Music Technology Group - Universitat Pompeu Fabra
+#
+# This file is part of Essentia
+#
+# Essentia is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Affero General Public License as published by the Free
+# Software Foundation (FSF), either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the Affero GNU General Public License
+# version 3 along with this program. If not, see http://www.gnu.org/licenses/
+
+
+from essentia_test import *
+from numpy import mean, array, float32, square
+from pathlib import Path
+
+
+class TestAudio2Midi(TestCase):
+    def testEmpty(self):
+        self.assertComputeFails(Audio2Midi(), [])
+
+    def testZero(self):
+        pitch, loudness, message_type, midi_note, time_compensation = Audio2Midi()(
+            zeros(32)
+        )
+        self.assertEqual(pitch, 0)
+        self.assertEqual(loudness, 0.0)
+        self.assertEqual(message_type, [])
+        self.assertEqual(midi_note.tolist(), array([], dtype=float32).tolist())
+        self.assertEqual(time_compensation.tolist(), array([], dtype=float32).tolist())
+
+    def assessNoteList(
+        self,
+        reference_path: str,
+        estimated: list,
+        n_notes_tolerance: int = 0,
+        onset_tolerance: float = 0.01,
+        offset_tolerance: float = 0.01,
+        midi_note_tolerance: int = 0,
```

Comment:
This can be hardcoded inside asserts directly

Reply:
No, it cannot bacause it is inside runARealCase() which receive different values for those parameters. If we hardcoded some tests will fail. I can remove the default value, that would be okay.

Reply:
Please resolve or suggest any others changes.